app/scripts/controllers/mobileClients.js

```diff
@@ -0,0 +1,86 @@
+'use strict';
+
+/**
+ * @ngdoc function
+ * @name openshiftConsole.controller:MobileClientsController
+ * @description
+ * Controller of the openshiftConsole
+ */
+angular.module('openshiftConsole')
+  .controller('MobileClientsController',
+    function ($scope,
+              $filter,
+              $routeParams,
+              Constants,
+              DataService,
+              ProjectsService) {
+
+      $scope.projectName = $routeParams.project;
+      $scope.emptyMessage = "Loading...";
+      $scope.alerts = {};
+      $scope.imagesByDockerReference = {};
+      $scope.redirectUrl = "/project/" + $scope.projectName + "/overview";
+      $scope.breadcrumbs = [
+        {
+          title: "Mobile Clients",
+          link: "project/" + $scope.projectName + "/browse/mobile-clients"
+        },
+        {
+          title: $routeParams.mobileclient
+        }
+      ];
+
+      var watches = [];
+
+      ProjectsService
+        .get($routeParams.project)
+        .then(_.spread(function(project, context) {
+          $scope.project = project;
+          $scope.projectContext = context;
+
+          DataService.get(Constants.MOBILE_CLIENT_VERSION, $routeParams.mobileclient, context, { errorNotification: false }).then(
+            // success
+            function(mobileClient) {
+              $scope.loaded = true;
+              $scope.mobileClient = mobileClient;
+
+              // If we found the item successfully, watch for changes on it
+              watches.push(DataService.watchObject(Constants.MOBILE_CLIENT_VERSION, $routeParams.mobileclient, context, function(mobileClient, action) {
+                if (action === "DELETED") {
+                  $scope.alerts["deleted"] = {
+                    type: "warning",
+                    message: "This mobile client has been deleted."
+                  };
+                }
+                $scope.mobileClient = mobileClient;
+                $scope.sdkConfigDoc = getSdkConfigDoc(mobileClient.spec.clientType);
+
+              }));
+            },
+            // failure
+            function(e) {
+              $scope.loaded = true;
+              $scope.alerts["load"] = {
+                type: "error",
+                message: e.status === 404 ? "This mobile client can not be found, it may have been deleted." : "The mobile client details could not be loaded.",
+                details: $filter('getErrorDetails')(e)
+              };
+            }
+          );
+
+          $scope.$on('$destroy', function(){
+            DataService.unwatchAll(watches);
+          });
+        }));
+
+      var getSdkConfigDoc = function(clientType) {
```

Comment:
If you make the templates into components we can just do a switch on the `clientType` in the html and have a default html for no `clientType`.

Craig also uses [similar logic](https://github.com/openshift/origin-web-console/commit/2b3a861b678166afba72407baeb2bf017f87c609#diff-4cc5d017ab8a77ecfad04db4f53e2066R35) so maybe a filter too?

```.html
<div ng-switch="mobileClient | mobileType">
  <ios-sdk-config ng-switch-when="iOS">...</ios-sdk-config>
  <android-sdk-config ng-switch-when="android">...</android-sdk-config>
  <div ng-switch-default>
    <article>
      <h2>Getting started</h2>
      <p>SDK is under construction.
        No SDK available at the moment.</p>
    </article>
  </div>
</div>
```

Probably looking at constants for the clientTypes too

Reply:
I've opted for dumb html templates as maintainability of this documentation is still undecided, I would love to have these files build/pulled during console build process.
+1 on default case

Reply:
yep we need to discuss how to manage the template content.

How about?
```.html
<div ng-switch="mobileClient | mobileType">
  <ng-include ng-switch-when="iOS" src="views/mobile-client-sdk-config/sdk-config-ios.html"></ng-include>
  <ng-include ng-switch-when="android" src="views/mobile-client-sdk-config/sdk-config-android.html"></ng-include>
  <div ng-switch-default>
    <article>
      <h2>Getting started</h2>
      <p>SDK is under construction.
        No SDK available at the moment.</p>
    </article>
  </div>
</div>
```